QueueDown/PipesAndChannels5.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System.Buffers;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.IO.Pipelines;
+using System.Threading.Channels;
+
+partial class Program
+{
+    // for debugging
+    static Output[] s_outputs = new Output[50];
+
+    public static PipeReader Pipes5(Counter<long> counter, List<Task> tasks)
+    {
+        //var pool = new PinnedBlockMemoryPool();
+        var pool = MemoryPool<byte>.Shared;
+        var channel = Channel.CreateBounded<Output>(new BoundedChannelOptions(50) 
+        {
+            SingleReader = true
+        });
+
+        // Multiple producers
+        for (int i = 0; i < 50; i++)
+        {
+            var producerPipe = new Pipe(new(pool, minimumSegmentSize:64*1024 ));
+            var output = new Output { Reader = producerPipe.Reader, Id = i.ToString() };
+            s_outputs[i] = output;
+            var b = (byte)i;
+
+            var producer1 = Task.Run(async () =>
+            {
+                // Write to the pipe
+                while (true)
+                {
+                    var buffer = producerPipe.Writer.GetMemory();
+                    int writeSize = 4096;
+                    buffer.Span.Slice(0, writeSize).Fill(b);
```

Comment:
This works but it's cheating a little. You need to ask for 4K if you want 4K (as an arg to GetMemory)